src/ModelContextProtocol/Client/McpClient.cs

```diff
@@ -39,10 +39,9 @@ public McpClient(IClientTransport clientTransport, McpClientOptions options, Mcp
             {
                 throw new InvalidOperationException($"Sampling capability was set but it did not provide a handler.");
             }
-
             SetRequestHandler<CreateMessageRequestParams, CreateMessageResult>(
                 RequestMethods.SamplingCreateMessage,
-                (request, ct) => samplingHandler(request, ct));
+                (request, ct) => samplingHandler(request, new ClientTokenProgress(this, new()), ct));
```

Comment:
What is the purpose of this `new()` here? `ProgressToken` is a struct; `new()` is just creating a default `ProgressToken` that doesn't actually contain any progress value. Every progress message it raises will be invalid.